src/__swaps__/screens/Swap/resources/search/search.ts

```diff
@@ -30,13 +30,19 @@ export type TokenSearchArgs = {
   list: TokenSearchListId;
   threshold?: TokenSearchThreshold;
   query?: string;
+  shouldPersist?: boolean;
 };
 
 // ///////////////////////////////////////////////
 // Query Key
 
-const tokenSearchQueryKey = ({ chainId, fromChainId, keys, list, threshold, query }: TokenSearchArgs) =>
-  createQueryKey('TokenSearch', { chainId, fromChainId, keys, list, threshold, query }, { persisterVersion: 2 });
+const tokenSearchQueryKey = ({ chainId, fromChainId, keys, list, threshold, query, shouldPersist }: TokenSearchArgs) => {
+  return createQueryKey(
+    'TokenSearch',
+    { chainId, fromChainId, keys, list, threshold, query },
+    { persisterVersion: shouldPersist ? 3 : undefined }
```

Comment:
using `shouldPersist` to distinguish when we want to include `persisterVersion` or not in the query key (which is later used by the queryClient in the `shouldDehydrateQuery` for a per-query-basis control of what to persist or not) https://github.com/rainbow-me/rainbow/blob/develop/src/react-query/queryClient.ts#L44

Reply:
jxom with a nice line there